examples/pip_parse_vendored/requirements.bzl

```diff
@@ -7,47 +7,149 @@ from //:requirements.txt
+}
+
+_config = {
+    "download_only": False,
+    "enable_implicit_namespace_pkgs": False,
+    "environment": {},
+    "extra_pip_args": [],
+    "isolated": True,
+    "pip_data_exclude": [],
+    "python_interpreter": "python3",
+    "python_interpreter_target": interpreter,
+    "quiet": True,
+    "repo": "pip",
+    "repo_prefix": "pip_",
+    "timeout": 600,
+}
+
+_annotations = {
+}
+
+# buildifier: disable=function-docstring
 def _clean_name(name):
     return name.replace("-", "_").replace(".", "_").lower()
 
-def requirement(name):
-    return "@pip_" + _clean_name(name) + "//:pkg"
+# buildifier: disable=function-docstring
+def requirement(name, use_clusters = True):
+    cname = _clean_name(name)
+    if cname in _cluster_mappings and use_clusters:
+        return "@pip//:_cluster_{}_dep_{}".format(_cluster_mappings[cname], cname)
+    else:
+        return "@pip_" + cname + "//:pkg"
 
-def whl_requirement(name):
-    return "@pip_" + _clean_name(name) + "//:whl"
+# buildifier: disable=function-docstring
+def whl_requirement(name, use_clusters = True):
+    cname = _clean_name(name)
+    if cname in _cluster_mappings and use_clusters:
+        return "@pip//:whl_" + _cluster_mappings[cname]
+    return "@pip_" + cname + "//:whl"
 
+# buildifier: disable=function-docstring
 def data_requirement(name):
     return "@pip_" + _clean_name(name) + "//:data"
 
+# buildifier: disable=function-docstring
 def dist_info_requirement(name):
     return "@pip_" + _clean_name(name) + "//:dist_info"
 
+# buildifier: disable=function-docstring
 def entry_point(pkg, script = None):
     if not script:
         script = pkg
     return "@pip_" + _clean_name(pkg) + "//:rules_python_wheel_entry_point_" + script
 
+# buildifier: disable=function-docstring
 def _get_annotation(requirement):
     # This expects to parse `setuptools==58.2.0     --hash=sha256:2551203ae6955b9876741a26ab3e767bb3242dafe86a32a749ea0d78b6792f11`
     # down to `setuptools`.
     name = requirement.split(" ")[0].split("=")[0].split("[")[0]
     return _annotations.get(name)
 
+# buildifier: disable=function-docstring
 def install_deps(**whl_library_kwargs):
     whl_config = dict(_config)
     whl_config.update(whl_library_kwargs)
-    for name, requirement in _packages:
+
+    # Install normal requirements
+    for name, spec in _packages:
         whl_library(
             name = name,
-            requirement = requirement,
-            annotation = _get_annotation(requirement),
+            requirement = spec,
+            annotation = _get_annotation(spec),
             **whl_config
         )
+
+    # And deal with requirement_clusters
```

Comment:
Can you include more comments? Really spell out what "dealing" with requirement_clusters means.